lib/trento_web/auth/jwt_auth_plug.ex

```diff
@@ -0,0 +1,93 @@
+defmodule TrentoWeb.Auth.JWTAuthPlug do
+  @moduledoc """
+    The JWTAuthPlug is a Pow compatible authorization flow.
+    Handles the login and the credentials recovery at each request
+
+    Uses Joken for jwt management
+
+    See the pow documentation for further details.
+  """
+  use Pow.Plug.Base
+
+  require Logger
+
+  alias Plug.Conn
+  alias TrentoWeb.Auth.AccessToken
+  alias TrentoWeb.Auth.RefreshToken
+
+  @impl true
+  @spec fetch(Plug.Conn.t(), any) :: {Plug.Conn.t(), nil | %{optional(<<_::40>>) => binary}}
```

Comment:
the spec is funky and not needed since we have already an @impl true
I would anyway add a docstring here